controller/test/blower_fsm/blower_fsm_test.cpp

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "blower_fsm.h"
+
+#include "hal.h"
+#include "gtest/gtest.h"
+
+class BlowerFsmTest : public ::testing::Test {
+public:
+  BlowerFsmTest() = default;
+  void SetUp() override { blower_fsm_init(); }
+};
+
+TEST_F(BlowerFsmTest, InitiallyOff) {
+  VentParams p = VentParams_init_zero;
+  EXPECT_EQ(blower_fsm_get_setpoint(p), kPa(0));
+}
+
+TEST_F(BlowerFsmTest, StaysOff) {
+  VentParams p = VentParams_init_zero;
+  Hal.delay(1000);
+  EXPECT_EQ(blower_fsm_get_setpoint(p), kPa(0));
+}
+
+TEST_F(BlowerFsmTest, PressureControl) {
+  VentParams p = VentParams_init_zero;
+  p.mode = VentMode_PRESSURE_CONTROL;
+  // 20 breaths/min = 3s/breath.  I:E = 2 means 2s for inspire, 1s for expire.
+  p.breaths_per_min = 20;
+  p.inspiratory_expiratory_ratio = 2;
+  p.peep_cm_h2o = 10;
+  p.pip_cm_h2o = 20;
+
+  EXPECT_EQ(blower_fsm_get_setpoint(p).cmH2O(), 20);
```

Comment:
I'd structure this a little differently but fine as is too:

```
vector<tuple</*time_millis=*/int, /*setpoint=*/float>> expected =
    {{0, 20}, {1000, 20}, {1999, 20}, ...};
```
and iterate over it with sleeping for the delta between neighboring points.

Reply:
Great idea.  Factored it to make it easier for someone to contribute additional tests, too.

Reply:
Going to make this change in #217 because it's annoying to move down here.